test/test_main.py

```diff
@@ -287,43 +286,62 @@ def _assess_setup_hooks(event_spec):
 
             for event_hook in event_hooks:
                 self.assertTrue(os.path.exists(event_hook))
-                self.assertEqual(os.readlink(event_hook), 'install')
-                self.assertEqual(os.readlink('hooks/install'),
-                                 self.CHARM_PY_RELPATH)
+                self.assertEqual(os.readlink(event_hook), self.CHARM_PY_RELPATH)
 
         charm_config = base64.b64encode(pickle.dumps({
             'STATE_FILE': self._state_file,
         }))
 
-        # Assess 'install' first because upgrade-charm or other
-        # events cannot be handled before install creates symlinks for them.
-        events_to_assess = (
-            EventSpec(InstallEvent, 'install', charm_config=charm_config),
-            EventSpec(StartEvent, 'start', charm_config=charm_config),
-            EventSpec(ConfigChangedEvent, 'config-changed', charm_config=charm_config),
-            EventSpec(LeaderElectedEvent, 'leader-elected', charm_config=charm_config),
-            EventSpec(UpgradeCharmEvent, 'upgrade-charm', charm_config=charm_config),
-            EventSpec(UpdateStatusEvent, 'update-status', charm_config=charm_config),
-        )
-
-        for event_spec in events_to_assess:
-            _assess_setup_hooks(event_spec)
-
-        self._clear_symlinks()
+        test_cases = [(
+            [
+                # Assess 'install' first because upgrade-charm or other events cannot be handled before install creates symlinks for them.
+                EventSpec(InstallEvent, 'install', charm_config=charm_config),
+            ], [
+                EventSpec(StartEvent, 'start', charm_config=charm_config),
+                EventSpec(ConfigChangedEvent, 'config-changed', charm_config=charm_config),
```

Comment:
Per our conversation, this test needs to be reworked. It became too convoluted and copy and pasted for us to understand what it's doing anymore.